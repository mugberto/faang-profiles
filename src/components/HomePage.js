import Grid from "@material-ui/core/Grid";
import logo from '../images/logo.png';

const HomePage = () => {
  const firms = [
    {
      symbol: 'fb',
      name: 'Facebook',
      price: 382,
    },
    {
      symbol: 'AAPL',
      name: 'Apple',
      price: 156,
    },
    {
      symbol: 'AMZN',
      name: 'Amazon',
      price: 3509,
    },
    {
      symbol: 'NFLX',
      name: 'Netflix',
      price: 606,
    },
    {
      symbol: 'GOOG',
      name: 'Google',
      price: 2885,
    },

  ];
  return (
    <Grid container spacing={0}>
      <Grid item xs={12}>
        <img src={logo} alt="logo" />
      </Grid>
      {firms.map((firm) => (<Grid item xs={6} key={firm.symbol}>
        <div>{firm.name}</div>
        <div>{firm.price}</div>
      </Grid>))}
    </Grid>
  );
};

export default HomePage;