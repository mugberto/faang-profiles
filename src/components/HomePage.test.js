import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen, fireEvent } from '../test-utils';
import HomePage from './HomePage';

describe('HomePage', () => {
  it('renders page correctly', () => {
    const tree = render(<HomePage />);
    expect(tree).toMatchSnapshot();
  });

  it('updates the UI with loaded data is loaded', async () => {
    render(<Router><HomePage /></Router>);
    expect(await screen.findByText('PROFILE')).toBeInTheDocument();
    expect(await screen.findByText('Comcast Corporation')).toBeInTheDocument();
  });
});
