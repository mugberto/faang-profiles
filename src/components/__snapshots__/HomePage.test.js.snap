// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root makeStyles-hero-1 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-6"
            style="padding-right: 8px;"
          >
            <img
              alt="logo"
              class="makeStyles-logo-7"
              src="logo.png"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-6"
            style="padding-left: 8px;"
          >
            <h1
              style="line-height: 1; font-size: 1.2rem;"
            >
              FIRM
              <br />
              <span
                style="margin-left: 0.7em;"
              >
                PROFILE
              </span>
            </h1>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-boxTitle-5 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <form
            class="inner-form-container"
          >
            <input
              class="searchField"
              name="name"
              placeholder="Search Company"
              type="text"
            />
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-hero-1 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-6"
          style="padding-right: 8px;"
        >
          <img
            alt="logo"
            class="makeStyles-logo-7"
            src="logo.png"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-6"
          style="padding-left: 8px;"
        >
          <h1
            style="line-height: 1; font-size: 1.2rem;"
          >
            FIRM
            <br />
            <span
              style="margin-left: 0.7em;"
            >
              PROFILE
            </span>
          </h1>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-boxTitle-5 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <form
          class="inner-form-container"
        >
          <input
            class="searchField"
            name="name"
            placeholder="Search Company"
            type="text"
          />
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
